src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -692,11 +728,11 @@ internal void RemoveAllKeys()
             }
 
             /// <summary>Removes the specified key from the table, if it exists.</summary>
-            internal bool Remove(TKey key)
+            internal bool Remove(TKey key, out object? value)
```

Comment:
Why is this `out object? value` rather than `[MaybeNullWhen(false)] out TValue value`?

Reply:
I didn't want to add the cast in the internal helper to avoid potentially extra overhead for the local even in cases callers would discard it anyway. But I assume the JIT would be smart enough anyway. I've updated the signature and callsites.